libmscore/score.cpp

```diff
@@ -3252,6 +3252,12 @@ void Score::collectMatch(void* data, Element* e)
                   return;
             }
 
+      if (p->measure && (p->measure != e->findMeasure()))
+            return;
+
+      if ((p->beat != -1.0) && (p->beat != e->beat()))
```

Comment:
Comparing float's is always a risky thing. Better solution to make beat a Fraction in stead of a float?

Reply:
Seems to work fine for me and don't see a need to use numerator/denominator. This uses the same value as with the status bar makeup.

Reply:
Look at the binary representation of a '-1.0' in IEEE float. Even if that can be represented as an exact number, what happens when beat is -1.0000000009? Should that be treated as equal or as not equal?

Reply:
I don't doubt it works. At least in most cases and that's the problem. The reason is two different floating point calculations will not necessarily come up with exact the same floating point result. See, among others, [Why not use Double or Float to represent currency?](https://stackoverflow.com/questions/3730019/why-not-use-double-or-float-to-represent-currency).
So, in most cases there is no problem but it is an accident to happen.

Reply:
Not sure I understand. A beat can't be -1.0, let alone -1.00009 as I understand it. As a matter of fact, I was tempted to use 0.0 for this, but to go beyond all doubt since wasn't sure if the beat could be 0.0 internally ever under some weird circumstance, I used -1.0.

Reply:
It is not only about the -1.0 but also the expression `p->beat != e->beat()`. The change this expression returns false is quite big even when `p->beat` and `e->beat()` are meant to have the same value. E.g. 4/3 can not be mapped onto an IEEE floating point number and will be something like 1.333333332 (or .333333334). The result depends on how the result is calculated. 4/3 might give another floating point result as 8/6 or 1+1/3.

Reply:
At any rate, it looks like the mtest is doing something funky. 
tst_copypastesymbollis failed.... any ideas there?
And again, give me a little time and i'll try this out with Fraction and do a recommit if it is working fine (don't see why it wouldn't). Don't want to risk having something not work that I'm not seeing based on testing so far.

Reply:
> @Jojo-Schmitz: Not sure I understand. A beat can't be -1.0, let alone -1.00009 as I understand it. 
So why testing for something that can't ever happen? You could simply use `false` instead ;-)

Reply:
P.S. Thanks for the concern of having this work without flaw.

Reply:
> P.S. Thanks for the concern of having this work without flaw.
No worries, I spend too much of my working life to solve these kind of issues and I'm glad if I can spare somebody this troubel :-)

Reply:
-1.0 is set up to indicate that [beat] was't set in the Note/Element pattern. If it is -1.0, there's no need to calculate the beat of the current element/note being tested, so it's a little more efficient this way than directly doing a != comparison to the beat of the element first, and i'm sure floats can't take boolean values.

Reply:
`if ((false) && (p->beat != e->beat()))`:-)

Reply:
Yeah I think Fraction has isValid() which returns true if the numerator is 0. Again, supposing there can't be a 0 beat (I can't imagine there being a 0 beat), then that will work just fine.